### PR TITLE
feat: add update user API

### PR DESCRIPTION
**Pull request by rishusingh022**
done with update user endpoint with 100 percent user services and controller test

### CODE REVIEW

File: insertdb.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const { users } = require('./src/models');
```
Comment (rohini-senthil) [inline insertdb.js:1]: Is this file committed by mistake?

File: src/index.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
 //create an express server 
 const express = require('express');
 const router = require('./routes/user.routes');
-const authRouter = require('./routes/loginRoutes');
-require('dotenv').config();
-const projectRouter = require('./routes/projectRoutes');
+// const authRouter = require('./routes/loginRoutes');
```
Comment (rohini-senthil) [inline src/index.js:4]: Can we uncomment this?

File: src/controllers/user.controller.js [hunk 3]
```diff
@@ -7,5 +7,18 @@ const getUsers = async (_, res) => {
...
+    const updatedUser = await userServices.updateUser(id, body);
+    if(!updatedUser) res.status(404).json({ message: 'User not found' });
+    res.status(200).json(updatedUser);
+  }
+  catch (error) {
+    res.status(500).json({
+      message: 'Something went wrong',
+    });
+  }
+};
```
Comment (Chetan-MckinseyBEBR) [inline src/controllers/user.controller.js:23]: Give new line before export

File: src/services/user.service.js [hunk 4]
```diff
@@ -5,7 +5,17 @@ const userServices = {
...
+      user[key] = userDetails[key];
+    }
+    await user.save();
+    return user;
   }
 };
 
-module.exports = userServices;
\ No newline at end of file
+module.exports = userServices;
```
Comment (rohini-senthil) [inline src/services/user.service.js:19]: Can we export this as 
```
module.exports = { updateUser, getAllUsers };
```

File: src/middlewares/update.user.validator.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+const Joi = require('joi');
```
Comment (Chetan-MckinseyBEBR) [inline src/middlewares/update.user.validator.js:1]: Put and empty line after a code block. Format this file

File: src/routes/user.routes.js [hunk 6]
```diff
@@ -1,10 +1,13 @@
 
 const express = require('express');
-const { getUsers } = require('../controllers/user.controller');
+const { getUsers , updateUser } = require('../controllers/user.controller');
+const {updateValidator} = require('../middlewares/update.user.validator');
```
Comment (Chetan-MckinseyBEBR) [inline src/routes/user.routes.js:4]: imports are not consistently formatted

File: insertdb.js (hunk 1 above)
Comment (rishusingh022) [reply insertdb.js:1]: yes

File: src/services/user.service.js (hunk 4 above)
Comment (rishusingh022) [reply src/services/user.service.js:19]: sure !!

File: src/index.js (hunk 2 above)
Comment (rishusingh022) [reply src/index.js:4]: yes sure !!

**Review by Chetan-MckinseyBEBR — COMMENTED** [review]
- Need code formatting at multiple places
- Plz resolve the comments

File: src/routes/user.routes.js (hunk 6 above)
Comment (rishusingh022) [reply src/routes/user.routes.js:4]: looking into it

File: src/services/user.service.js (hunk 4 above)
Comment (rishusingh022) [reply src/services/user.service.js:19]: actually it cannot be as userServices is an object it was pulled from previous pr of get all user so i cannot do like that

File: src/routes/projectRoutes.js [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
+const {getProject,listProjects,createProject,deleteProject,updateProject}=require('../controllers/project.controller');
 
 
 const projectRouter=express.Router();
 
 
-projectRouter.route('/')
-  .get(listProjects)
-  .post(createProject);
+// projectRouter.route('/')
```
Comment (rohini-senthil) [inline src/routes/projectRoutes.js:8]: Let's remove these lines if they're not implemented yet

**Pull request closed without merging**